src/lib/utils.ts

```diff
@@ -86,10 +86,15 @@ export const screenshotElement = (element: HTMLElement): Promise<Blob> => {
 
 // Make sure image's size is within 20MB and 2048x2048 resolution
 // https://platform.openai.com/docs/guides/vision/is-there-a-limit-to-the-size-of-the-image-i-can-upload
-export const compressImageToBase64 = (file: File): Promise<string> => {
+export const compressImageToBase64 = (
+  file: File,
+  compressionFactor: number = 1,
+  maxSizeMB: number = 20,
+  maxWidthOrHeight: number = 2048
```

Comment:
```suggestion
  compressionFactor: number = getSettings().compressionFactor,
  maxSizeMB: number = getSettings().maxCompressedFileSizeMB,
  maxWidthOrHeight: number = getSettings().maxImageDimension
```

Instead of hard-coding default values, shouldn't we use the current settings values?
This would eliminate the need to pass settings.xyz explicitly wherever you are currently using these. 

But you might also want to keep it decoupled from settings state, as you are doing right now, so its a personal preference I guess.

Also, shouldn't we rather use an options object instead of flat args. For cases where I want to pass only one option, currently I will be forced to pass other args as well.

Reply:
The options object idea is great! Let me fix that. 
For the hard-coding default values, I prefer to keep it decoupled from settings as it's in the `util`